src/com/geekhub/hw2/cars/SolarCar.java

```diff
@@ -0,0 +1,24 @@
+package com.geekhub.hw2.cars;
+
+import com.geekhub.hw2.parts.interfaces.EnergySource;
+import com.geekhub.hw2.parts.interfaces.Engine;
+import com.geekhub.hw2.parts.interfaces.Gear;
+import com.geekhub.hw2.parts.interfaces.RunMechanism;
+
+public class SolarCar extends Car {
+
+    public SolarCar(EnergySource energySource, Engine engine, Gear gear, RunMechanism runMechanism) {
```

Comment:
In this case better to pass concrete implementation as pararameters. 
E.g. Now I can do something like that: "new SolarCar(new GasTank(3, 4), new CarEngine(3f), new BoatTransmission(new Wheels(17)), new Screw());"